src/test/java/JsonUtilityTest.java

```diff
@@ -12,94 +12,138 @@
+  private static final CustomClass CUSTOM_OBJECT = CustomClass.getInstance("shhh");
+
+  private static HttpServletResponse response;
+  private static StringWriter stringWriter;
+  private static PrintWriter printWriter;
+  private final Gson gson = new Gson();
+
   @Before
-  public void flush() {
-    // Flush in case text is still in buffer
-    stringWriter.flush();
+  public void setUp() throws IOException {
+    // Mock servlet response writer
+    response = Mockito.mock(HttpServletResponse.class);
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    Mockito.when(response.getWriter()).thenReturn(printWriter);
   }
 
   @After
-  public void clear() {
+  public void tearDown() throws IOException {
     // Dump contents after each test
     stringWriter.getBuffer().setLength(0);
+    stringWriter.close();
+    printWriter.close();
   }
 
   @Test
   public void sendStringToResponse() throws IOException {
-    // String should be sent successfully and in the expected format
+    // Sends an object of type String to response as json.
+    // Response should have the same contents as String object.
     JsonUtility.sendJson(response, STRING_OBJECT);
+    printWriter.flush();
 
-    String expected = gson.toJson(STRING_OBJECT);
-    String actual = stringWriter.toString();
+    String actual = gson.fromJson(stringWriter.toString(), String.class);
 
-    Assert.assertTrue(actual.contains(expected));
+    Assert.assertEquals(STRING_OBJECT, actual);
   }
 
   @Test
   public void sendListToResponse() throws IOException {
-    // List should be sent successfully and in the expected format
+    // Sends an object of type List to response as json.
+    // Response should have the same contents as List object.
     JsonUtility.sendJson(response, LIST_OBJECT);
+    printWriter.flush();
 
-    String expected = gson.toJson(LIST_OBJECT);
-    String actual = stringWriter.toString();
+    String actualString = stringWriter.toString();
+    Type type = new TypeToken<List<Integer>>() {}.getType();
+    List<Integer> actual = gson.fromJson(actualString, type);
 
-    Assert.assertTrue(actual.contains(expected));
+    Assert.assertEquals(LIST_OBJECT, actual);
   }
 
   @Test
   public void sendMapToResponse() throws IOException {
-    // Map should be sent successfully and in the expected format
+    // Sends an object of type Map to response as json.
+    // Response should have the same contents as Map object.
     JsonUtility.sendJson(response, MAP_OBJECT);
+    printWriter.flush();
+
+    String actualString = stringWriter.toString();
+    Type type = new TypeToken<Map<String, Integer>>() {}.getType();
+    Map<String, Integer> actual = gson.fromJson(actualString, type);
+
+    Assert.assertEquals(MAP_OBJECT.get("One"), actual.get("One"));
+    Assert.assertEquals(MAP_OBJECT.get("Two"), actual.get("Two"));
+    Assert.assertEquals(MAP_OBJECT.get("Three"), actual.get("Three"));
```

Comment:
I think `Assert.assertEquals(MAP_OBJECT, actual)` should do the right thing. Is there a reason you don't use it?

Reply:
Thanks for pointing this out! Not sure why I underestimated `assertEquals`.

Reply:
Yeah, `Assert.assertEquals` just uses `object.Equals`, which is correctly defined for `Map`.

Reply:
Nice!